source/util/FileUtils.cpp

```diff
@@ -66,7 +66,11 @@ string FileUtils::ExtractExpandedPath(const string &filePath)
         return "";
     }
     wordexp_t word;
-    wordexp(filePath.c_str(), &word, 0); // TODO: How to handle error?
+    int status = wordexp(filePath.c_str(), &word, 0); // TODO: How to handle error?
+    if (status == WRDE_BADCHAR)
+    {
+        return "-1";
```

Comment:
Checking the status and returning a sentinel value prevents the segmentation fault, but since no callers of this method are prepared to check for the returned sentinel this doesn't really address the problem. Furthermore, the sentinel value is potentially ambiguous since "-1" is an unlikely but valid filename path. Instead of returning sentinel value, I recommend that you throw a `std::runtime_error` [1] (or an error with a custom name that inherits from runtime_error as shown in [2]).  The exception message should include the offending `filePath`.

The vast majority of instances in which `FileUtils::ExtractExpandedPath` is called emanate from `Config::init`. As a result, you can put the exception handler in that function. The handler should log the exception message with `LOGM_ERROR` and return `false` from the function. This will fail the device client on startup.

We do have a few instances where `FileUtils::ExtractExpandedPath` is called after config is loaded and while a feature is started. Examples include `SampleShadowFeature.cpp` and `PubSubFeature.cpp`.  It would be better if these feature implementations did the validation as part of `Config::init` so that we can fail the device client while parsing/validating config rather than at feature startup. If that approach isn't possible then you should add try-catch in each feature or possibly as part of `SharedCrtResourceManager::startDeviceClientFeatures` (not sure which is better). 

[1] https://en.cppreference.com/w/cpp/error/runtime_error

[2]
```
struct my_error : std::runtime_error 
{
    using std::runtime_error::runtime_error;
};
```